Go-file variant panics when fetching presumably missing code

When running
```
./build/trace run-vm --validate --db-impl carmen --db-variant go-file --
updatedir /<path_to>/updateset --substatedir /<path_to>/substate.50M  4564026 4600000
```
the `go-file` implementation panics with an error indicating an issue with the code fetching in files:
```
panic: Unable to obtain code for [29 240 61 161 98 221 2 167 208 41 192 165 25 218 82 139 161 205 229 141]: EOF

goroutine 1 [running]:
github.com/Fantom-foundation/Carmen/go/state.(*stateDB).GetCode(0xc0002000e0, {0x1d, 0xf0, 0x3d, 0xa1, 0x62, 0xdd, 0x2, 0xa7, 0xd0, ...})
        /home/wawa/dev/aida/carmen/go/state/state_db.go:432 +0x1ec
github.com/Fantom-foundation/Aida/tracer/state.(*carmenStateDB).GetCode(0x156127c691a7ecb5?, {0x1d, 0xf0, 0x3d, 0xa1, 0x62, 0xdd, 0x2, 0xa7, 0xd0, ...})
        /home/wawa/dev/aida/tracer/state/carmen.go:109 +0x37
github.com/Fantom-foundation/Aida/cmd/trace-cli/trace.validateStateDB(0x0?, {0xca1ca8, 0xc0000129f0})
        /home/wawa/dev/aida/cmd/trace-cli/trace/validate.go:29 +0x1be
github.com/Fantom-foundation/Aida/cmd/trace-cli/trace.runVMTask({0xca1ca8, 0xc0000129f0}, 0xc000b38080, 0x34?, 0x0?, 0x0?, {0xb93825?, 0x20?})
        /home/wawa/dev/aida/cmd/trace-cli/trace/run_vm.go:100 +0x30f
github.com/Fantom-foundation/Aida/cmd/trace-cli/trace.runVM(0x106d200?)
        /home/wawa/dev/aida/cmd/trace-cli/trace/run_vm.go:313 +0xced
github.com/urfave/cli/v2.(*Command).Run(0x106d200, 0xc000241040)
        /home/wawa/go/pkg/mod/github.com/urfave/cli/v2@v2.19.2/command.go:177 +0x746
github.com/urfave/cli/v2.(*App).RunContext(0xc000182700, {0xc98d88?, 0xc0000260b0}, {0xc000024340, 0xd, 0xd})
        /home/wawa/go/pkg/mod/github.com/urfave/cli/v2@v2.19.2/app.go:389 +0x105e
github.com/urfave/cli/v2.(*App).Run(...)
        /home/wawa/go/pkg/mod/github.com/urfave/cli/v2@v2.19.2/app.go:254
main.main()
        /home/wawa/dev/aida/cmd/trace-cli/main.go:33 +0x13b
```

--- a/go/backend/depot/depot_test.go
+++ b/go/backend/depot/depot_test.go
@@ -136,8 +136,11 @@ func TestSetToArbitraryPosition(t *testing.T) {
 				t.Fatalf("failed to set C; %s", err)
 			}
 
-			if value, _ := d.Get(1); value != nil {
-				t.Errorf("not-existing value is not reported as not-existing")
+			if value, err := d.Get(1); err != nil || value != nil {
+				t.Errorf("non-existing value is not reported as non-existing")
+			}
+			if value, err := d.Get(50); err != nil || value != nil {
+				t.Errorf("non-existing value is not reported as non-existing")
 			}
 			if value, _ := d.Get(5); !bytes.Equal(value, A) {
 				t.Errorf("reading written A returned different value")

--- a/go/backend/depot/file/file.go
+++ b/go/backend/depot/file/file.go
@@ -144,15 +144,12 @@ func (m *Depot[I]) Get(id I) (out []byte, err error) {
 	var offsetBytes [OffsetSize + LengthSize]byte
 	_, err = m.offsetsFile.ReadAt(offsetBytes[:], itemPosition)
 	if err != nil {
-		return nil, err
-	}
-	offset, length := parseOffsetLength(offsetBytes[:])
-	if err != nil {
 		if errors.Is(err, io.EOF) {
 			return nil, nil
 		}
 		return nil, err
 	}
+	offset, length := parseOffsetLength(offsetBytes[:])
 	if length == 0 {
 		return nil, nil
 	}

